core/common/src/TimeZone.kt

```diff
@@ -14,28 +14,58 @@ import kotlinx.serialization.Serializable
 /**
  * A time zone, provides the conversion between [Instant] and [LocalDateTime] values
  * using a collection of rules specifying which [LocalDateTime] value corresponds to each [Instant].
+ *
+ * A time zone can be used in [Instant.toLocalDateTime] and [LocalDateTime.toInstant], and also in
+ * those arithmetic operations on [Instant] that require knowing the calendar.
+ *
+ * A [TimeZone] can be constructed using the [TimeZone.of] function, which accepts the string identifier, like
+ * `"Europe/Berlin"`, `"America/Los_Angeles"`, etc. For a list of such identifiers, see [TimeZone.availableZoneIds].
+ * Also, the constant [TimeZone.UTC] is provided for the UTC time zone.
+ *
+ * For interaction with `kotlinx-serialization`, [TimeZoneSerializer] is provided that serializes the time zone as its
+ * identifier.
+ *
+ * @sample kotlinx.datetime.test.samples.TimeZoneSamples.usage
  */
 @Serializable(with = TimeZoneSerializer::class)
 public expect open class TimeZone {
     /**
      * Returns the identifier string of the time zone.
      *
      * This identifier can be used later for finding this time zone with [TimeZone.of] function.
+     *
+     * @sample kotlinx.datetime.test.samples.TimeZoneSamples.id
      */
     public val id: String
 
-    // TODO: Declare and document toString/equals/hashCode
+    /**
+     * Equivalent to [id].
+     *
+     * @sample kotlinx.datetime.test.samples.TimeZoneSamples.equalsSample
+     */
+    public override fun toString(): String
+
+    /**
+     * Compares this time zone to the other one. Time zones are equal if their identifier is the same.
+     *
+     * @sample kotlinx.datetime.test.samples.TimeZoneSamples.equalsSample
+     */
+    public override fun equals(other: Any?): Boolean
 
     public companion object {
         /**
          * Queries the current system time zone.
          *
          * If the current system time zone changes, this function can reflect this change on the next invocation.
+         *
+         * @sample kotlinx.datetime.test.samples.TimeZoneSamples.currentSystemDefault
          */
         public fun currentSystemDefault(): TimeZone
 
         /**
          * Returns the time zone with the fixed UTC+0 offset.
+         *
+         * @sample kotlinx.datetime.test.samples.TimeZoneSamples.utc
```

Comment:
Can we document that `TimeZone.UTC != TimeZone.of("etc/UTC")` https://github.com/Kotlin/kotlinx-datetime/issues/222